### DISCUSSION TITLE
H.264 --> H.265

### DISCUSSION THREAD

**Discussion by mateothegreat** [bluenviron/mediamtx#3034] [Questions]
### Question

Does the framework support going from H.264 --> H.265 without needing to invoke ffmpeg but use the golang bindings directly?

I dug through the code and couldn't reach a conclusion.

I ask because the compute requirements are way higher vs. using the CGO bindings for ffmpeg.

If not, what would it take to get this in place? 💸 
Thanks!

**Comment by aler9** [comment]
Hello, native re-encoding is not present and will not be developed for various reasons.

From a user point of view, discouraging re-encoding allow users to save a huge amount of CPU and RAM.

From a technical point of view, CGO bindings strongly decrease portability of the software: on Linux they make the executable dependent on a specific version of libc, and therefore on a specific distribution and OS version, on Windows the issue is even greater as we would need to ship the server together with libav, increasing size by 10 times. Furthermore, there are various ways to perform both decoding and encoding (hardware-based, software-based), and covering them all would be impossible. Furthermore, performance differences between native bindings and external calls to FFmpeg are minimal, the great majority of computational power is absorbed by the encoding process and not by I/O.

From a legal point of view, decoding and encoding of H264 and H265 are processes that are covered by patent pools, which require software authors to pay fees. If the user chooses to perform decoding or encoding of such codecs, it has to use an external software.

**Comment by mateothegreat** [comment]
Makes sense and I totally agree, was more curious.
Thank you for the detailed answer @aler9!

[a comment by an automated account is not shown]
